pkg/controller/controller.go

```diff
@@ -111,6 +111,18 @@ func (c *controller) serviceBindingCallback(e k8swatch.Event) error {
 			return err
 		}
 		glog.Infof("Created Service Binding: %s\n%v\n", sb.Name, created)
+	} else if e.Type == k8swatch.Deleted {
+		// if we've already processed the deletion event, ignore
+		if sb.DeletionTimestamp != nil {
+			return nil
+		}
+
+		if err := c.handler.DeleteServiceBinding(&sb); err != nil {
+			return err
+		}
+	} else if e.Type == k8swatch.Modified && sb.DeletionTimestamp != nil {
```

Comment:
What are the chances where it's modified and the timestamp is nil? Does that fall through to the "WarningF" log below? What does that mean? Do we need to log that deletion timestamp as well?